SourceX/DiabloUI/diabloui.cpp

```diff
@@ -694,91 +720,94 @@ void Render(const UiScrollBar &ui_sb)
 	{
 		const SDL_Rect rect = UpArrowRect(ui_sb);
 		const int frame = static_cast<int>(scrollBarState.upArrowPressed ? ScrollBarArrowFrame_UP_ACTIVE : ScrollBarArrowFrame_UP);
-		DrawArt(rect.x, rect.y, ui_sb.arrow, frame, rect.w);
+		DrawArt(rect.x, rect.y, ui_sb->m_arrow, frame, rect.w);
 	}
 	{
 		const SDL_Rect rect = DownArrowRect(ui_sb);
 		const int frame = static_cast<int>(scrollBarState.downArrowPressed ? ScrollBarArrowFrame_DOWN_ACTIVE : ScrollBarArrowFrame_DOWN);
-		DrawArt(rect.x, rect.y, ui_sb.arrow, frame, rect.w);
+		DrawArt(rect.x, rect.y, ui_sb->m_arrow, frame, rect.w);
 	}
 
 	// Thumb:
-	{
+	if (SelectedItemMax - SelectedItemMin > 0) {
 		const SDL_Rect rect = ThumbRect(
 		    ui_sb, SelectedItem - SelectedItemMin, SelectedItemMax - SelectedItemMin + 1);
-		DrawArt(rect.x, rect.y, ui_sb.thumb);
+		DrawArt(rect.x, rect.y, ui_sb->m_thumb);
 	}
 }
 
-void Render(const UiEdit &ui_edit)
+void Render(const UiEdit *ui_edit)
 {
-	DrawSelector(ui_edit.rect);
-	SDL_Rect rect = ui_edit.rect;
+	DrawSelector(ui_edit->m_rect);
+	SDL_Rect rect = ui_edit->m_rect;
 	rect.x += 43;
 	rect.y += 1;
 	rect.w -= 86;
-	DrawArtStr(ui_edit.value, rect, ui_edit.flags, /*drawTextCursor=*/true);
+	DrawArtStr(ui_edit->m_value, rect, ui_edit->m_iFlags, /*drawTextCursor=*/true);
 }
 
-void RenderItem(UiItem *item)
+void RenderItem(UiItemBase *item)
 {
 	if (item->has_flag(UIS_HIDDEN))
 		return;
-	switch (item->type) {
+	switch (item->m_type) {
 	case UI_TEXT:
-		Render(&item->text);
+		Render((UiText *)item);
 		break;
 	case UI_ART_TEXT:
-		Render(item->art_text);
+		Render((UiArtText *)item);
 		break;
 	case UI_IMAGE:
-		Render(item->image);
+		Render((UiImage *)item);
 		break;
 	case UI_ART_TEXT_BUTTON:
-		Render(item->art_text_button);
+		Render((UiArtTextButton *)item);
 		break;
 	case UI_BUTTON:
-		RenderButton(&item->button);
+		RenderButton((UiButton *)item);
 		break;
 	case UI_LIST:
-		Render(item->list);
+		Render((UiList *)item);
 		break;
 	case UI_SCROLLBAR:
-		Render(item->scrollbar);
+		Render((UiScrollBar *)item);
 		break;
 	case UI_EDIT:
-		Render(item->edit);
+		Render((UiEdit *)item);
 		break;
 	}
 }
 
-bool HandleMouseEventArtTextButton(const SDL_Event &event, const UiArtTextButton &ui_button)
+bool HandleMouseEventArtTextButton(const SDL_Event &event, UiArtTextButton *ui_button)

```

Comment:
Is it necessary to remove `const` here (and throughout)?

Reply:
@marty28 I'm going to revert this, let me know if it's needed for VS2003

Reply:
@glebm at least `Render(UiText/UiList` have issues with using `const` even on GCC:
```
SourceX/DiabloUI/diabloui.cpp:698:38: error: passing ‘const dvl::UiList’ as ‘this’ argument discards qualifiers [-fpermissive]
  698 |   SDL_Rect rect = ui_list->itemRect(i);
```

Reply:
The `itemRect` function should be marked `const`